pkg/util/purge/resourcegroups.go

```diff
@@ -0,0 +1,126 @@
+package purge
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"sort"
+
+	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
+)
+
+// CleanResourceGroups loop through the resourgroups in the subscription
+// and deleted everything that is not marked for deletion
+// The deletion check is performed by passed function: DeleteCheck
+func (rc *ResourceCleaner) CleanResourceGroups(ctx context.Context) error {
```

Comment:
Nit: I would keep it together with the `ResourceCleaner` type definition. Not worth splitting into multiple fiels at this point.

Reply:
Soon we will add more files, so this is preparation.